docs/lib/graphqlapi/fragments/ios/advanced-workflows.md

```diff
@@ -0,0 +1,225 @@
+
+Then, query for the Todo by a todo id
+```swift
+Amplify.API.query(request: .getWithoutDescription(byId: "[UNIQUE_ID]")) { 
+  // handle result
+```  
+
+## Nested Data
+If you have a relational model, you can retrieve the nested object by creating a `GraphQLRequest` with a selection set containing the nested object's fields. For example, in this schema, the Post can contain multiple comments.
+```graphql
+enum PostStatus {
+  ACTIVE
+  INACTIVE
+}
+type Post @model {
+  id: ID!
+  title: String!
+  rating: Int!
+  status: PostStatus!
+  comments: [Comment] @connection(keyName: "byPost", fields: ["id"])
+}
+type Comment @model
+  @key(name: "byPost", fields: ["postID", "content"]) {
+  id: ID!
+  postID: ID!
+  post: Post! @connection(fields: ["postID"])
+  content: String!
+}
+```
+The `GraphQLRequest` will look like this
+
+```swift
+extension GraphQLRequest {
+    static func getPostWithComments(byId id: String) -> GraphQLRequest<JSONValue> {
+        let document = """
+        query getPost($id: ID!) {
+          getPost(id: $id) {
+            id
+            title
+            rating
+            status
+            comments {
+              items {
+                id
+                postID
+                content
+              }
+            }
+          }
+        }
+        """
+        return GraphQLRequest<JSONValue>(document: document,
+                                         variables: ["id": id],
+                                         responseType: JSONValue.self)
+    }
+}
+
+```
+Query with `Amplify.API.query(request: .getCommentWithPost(byId: "[COMMENT_ID]"))`. Notice here that `JSONValue` is used as the `responseType`. `JSONValue` is utility type that can be used to represent an arbitrary JSON response.
+
+## Combining Multiple Operations
+
+When you want to perform more than one operation in a single request, you can place them within the same document. For example, to retrieve a Post and a Todo
+```swift
+extension GraphQLRequest {
+    static func get(byPostId postId: String, todoId: String) -> GraphQLRequest<JSONValue> {
+        let document = """
+        query get($postId: ID!, $todoId: ID!) {
+          getPost(id: $postId) {
+            id
+            title
+            rating
+          }
+          getTodo(id: $todoId) {
+            id
+            name
+          }
+        }
+        """
+        return GraphQLRequest<JSONValue>(document: document,
+                                         variables: ["postId": postId,
+                                                     "todoId": todoId],
+                                         responseType: JSONValue.self)
+    }
+}
+```
+Once you have the response data in a `JSONValue`, you can access each object in the JSON structure by encoding it back to Data and decoding it to the expected Model.
```

Comment:
For Android, there is not a way to do this currently.   The deserializer will throw an exception if there is more than one element in the `data` object.    To add support for this, we could either:
1. If `data` has one element, deserialize into a `responseType` object (like we already do), but if more than one, deserialize into a `Map<{operationName}, {responseType}>`.
2. Add support for `decodePath` similar to iOS.

I'm inclined towards #1 because I'd prefer to avoid requiring customers to specify a `decodePath`.  I'm open to thoughts from you, because I think we should have platform parity.

Reply:
For 1.  I agree that when data only has one element, then the developer should be able to just specify the response type without having to set the decodePath to the operationName. the `decodePath` parameter isn't very useful when the data contains arrays, then you won't be able to go further than the objects before that, for example,
```
data {
  getTodo {
    id: 111
    items [
       {  id: 222 }, { id: 333 }
    ]
  }
  getPost {
    id: 444
  }
}
```
decodePath of "getTodo.items" is the furthest you can go since the value at "items" is an array of objects


when there is more than one, can we take no opinion on deserialization and just deserialize to whatever the response is at "data"? this will be an object like
```
ResponseType {
  operationName1: ResponseType1
  operationName2: ResponseType2
}
```
and would map to the same response structure that the developer can see if they use the console or tried deserialization to String and printed it out

Reply:
Ahh, that's right - #1 isn't actually possible since we would need the customer to provide the expected `responseType` for each operation, and currently, the GraphQLRequest constructor only asks for one `responseType`.

So, there's no way to support multiple operations with Android, currently. 

I know there has been at least one customer request for it, so I will keep thinking about the best way to support this, but for now, I think we should just say it is not supported in the docs.

Reply:
on iOS, we can first get parity in behavior: "if there is only one element at data, deserialize to it" as API behavior change, this will allow the documentation to be updated to omit the decodePath. 

for elements which is under "data.items", we'll need custom deserialization logic in the plugin code to support this like Android has. 

until both of these changes happen, APIPlugin on iOS requires the decodePath to be set for both scenarios
1. getting element under "data"
2. getting list of elements under "data.items" for list queries

Reply:
This PR on iOS should address the decodePath parity and then I can update the docs here to remove the `decodePath` examples https://github.com/aws-amplify/amplify-ios/pull/714

Reply:
getting list of elements under "data.items" has been merged in, however decodePath will go in later